src/core/Entry.cpp

```diff
@@ -501,6 +501,19 @@ QString Entry::totpSettingsString() const
     return {};
 }
 
+QString Entry::path() const
+{
+    QStringList h = this->group()->hierarchy();
+    h = h.mid(1);
+    QString path;
+    if (h.length() == 0) {
+        path = this->title();
+    } else {
+        path = h.join("/") + "/" + this->title();
+    }
+    return path;
+}
```

Comment:
@droidmonkey , this is executed with each request to `path()`. It can be refactored into a property but then updating etc would also need to be implemented. I do not believe that this operation is _that_ costly to justify the amount of extra code otherwise needed.

Reply:
We have a lot of lookup inefficiencies i am investigating for 2.7, especially related to searching.